.github/workflows/auto-dependabot.yml

```diff
@@ -0,0 +1,23 @@
+name: Dependabot auto-merge
+on: pull_request_target
+
+permissions:
+  pull-requests: write
+  contents: write
+
+jobs:
+  dependabot:
+    runs-on: ubuntu-latest
+    if: ${{ github.actor == 'dependabot[bot]' }}
+    steps:
+      - name: Dependabot metadata
+        id: metadata
+        uses: dependabot/fetch-metadata@v1.1.1
+        with:
+          github-token: "${{secrets.GITHUB_TOKEN}}"
+      - name: Enable auto-merge for Dependabot PRs
+        if: success()
```

Comment:
Correct me if I'm wrong, but the `success()` condition only applies to jobs where it is used. Wouldn't it be better to be sure that the compilation and unit tests are passed before merging on the master branch?

Reply:
This is what the check does. If the tests fail, dependabot will make a pull request stating that updating this dependency caused an error.